subsys/net/ip/ipv6.c

```diff
@@ -362,6 +362,28 @@ static inline enum net_verdict ipv6_route_packet(struct net_pkt *pkt,
 
 #endif /* CONFIG_NET_ROUTE */
 
+
+static enum net_verdict ipv6_forward_mcast_packet(struct net_pkt *pkt,
+						 struct net_ipv6_hdr *hdr)
+{
+#if defined(CONFIG_NET_ROUTE_MCAST)
+	/* check if routing loop could be created or if the destination is of
+	 * interface local scope or if from link local source
+	 */
+	if (net_ipv6_is_addr_mcast(&hdr->src)  ||
+	      net_ipv6_is_addr_mcast_iface(&hdr->dst) ||
+	       net_ipv6_is_ll_addr(&hdr->src)) {
+		return NET_CONTINUE;
+	}
+	int routed = net_route_mcast_forward_packet(pkt, hdr);
```

Comment:
Please declare the variable at the start of the function. Also please empty line after the }